src/JsonSchema/SchemaFactory.php

```diff
@@ -230,14 +233,17 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
                 break;
             }
 
-            if ($type->isNullable()) {
-                $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
-            } else {
-                $propertySchema['$ref'] = $subSchema['$ref'];
-            }
+            $refs[] = ['$ref' => $subSchema['$ref']];
+            $isNullable = $isNullable ?? $type->isNullable();
+        }
+
+        if ($isNullable && !\in_array(['type' => 'null'], $refs, true)) {
+            $refs[] = ['type' => 'null'];
+        }
 
+        if (\count($refs) > 0) {
```

Comment:
What if `\count($refs) === 1`?

Reply:
Same: what if `$refs` is empty?

Reply:
That's mildly embarrassing, but fixed now. :blush: